src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,355 @@
+        try {
+            String query = String.format("{\"query\": \"mutation { deleteCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(responseBody);
+            waitForDeletedCluster(clusterId, TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            LOG.info(String.format("Cluster with id %s is deleted", clusterId));
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new CloudException("Delete hazelcast cloud cluster is failed, error message is: " + e.getMessage());
+        }
+    }
+
+
+    private Response prepareAndSendRequest(String query) throws InterruptedException, CloudException {
+        int retryCountForExceptionOfEndpoint = 0;
+        // Rarely server returns empty header, that is why a retry mechanism is added.
+        while(retryCountForExceptionOfEndpoint < 3)
+        {
+            try {
+                RequestBody body = RequestBody.create(JSON, query);
+                Request request = new Request.Builder()
+                        .url(HttpUrl.get(uri))
+                        .post(body)
+                        .header("Authorization", String.format("Bearer %s", bearerToken))
+                        .header("Content-Type", "application/json")
+                        .build();
+
+                call = client.newCall(request);
+                return call.execute();
+            }
+            catch(Exception e)
+            {
+                Log.warn(e.toString());
+            }
+            retryCountForExceptionOfEndpoint++;
+            TimeUnit.SECONDS.sleep(2);
+        }
+        throw new CloudException("Request cannot send successfully after 3 tries");
+    }
+
+    private String getTlsPassword(String clusterId) {
+        String requestUrl = String.format("%s/cluster/%s", baseUrl, clusterId);
+        try {
+            Request request = new Request.Builder()
+                    .url(requestUrl)
+                    .get()
+                    .header("Authorization", String.format("Bearer %s", bearerToken))
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            try {
+                return mapper.readTree(response.body().string()).get("tlsPassword").asText();
+            } catch (Exception e) {
+                throw new CloudException("Body is null for tlsPassword");
+            }
+        }
+        catch(Exception e)
+        {
+            LOG.warn(e.getMessage());
+            return null;
+        }
+    }
+
+    private boolean waitForStateOfCluster(String clusterId, String expectedState, long timeoutInMillisecond) throws CloudException {
+        String currentState = "";
+        long startTime = System.currentTimeMillis();
+        int retryCycle = 10;
```

Comment:
Is this not too long 10secs?

Reply:
As the cluster creation and changing state takes around a few minutes, I think 10 sec is not too long. Also it is defined as a final global variable